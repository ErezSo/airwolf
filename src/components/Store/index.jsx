import React from "react";
import { Item } from "../";

const items = [
  { id: "h38ss-dtbl", img: "h38ss-dtbl-sel-201603.png" },
  { id: "h38ss-dtgr", img: "h38ss-dtgr-sel-201603.png" },
  { id: "h38ss-dtrd", img: "h38ss-dtrd-sel-201603.png" },
  { id: "h42ss-stbk", img: "h42ss-stbk-sel-201603.png" },
  { id: "s42gd-nsgb", img: "s42gd-nsgb-sel-201603.png" },
  { id: "s42rg-nsdb", img: "s42rg-nsdb-sel-201603.png" },
  { id: "s42sg-nsbk", img: "s42sg-nsbk-sel-201603.png" },
  { id: "s42si-nssb", img: "s42si-nssb-sel-201603.png" }
];

const styles = {
  container: {
    marginTop: 100
  },
  itemsList: {
    maxWidth: 1030,
    display: "flex",
    flexWrap: "wrap",
    margin: "0 auto",
    justifyContent: "center"
  }
};

function Store({ reserveItem }) {
  const { container, itemsList } = styles;
  return (
    <div style={container}>
      <div style={itemsList}>
        {items.map((item, idx) => (
          <Item item={item} key={idx} idx={idx} reserveItem={reserveItem} />
        ))}
      </div>
    </div>
  );
}

export default Store;
