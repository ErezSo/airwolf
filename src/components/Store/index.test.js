import React from "react";
import { shallow } from "enzyme";
import Store from "./";

describe("Store", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Store reserveItem={jest.fn()} />);
  });

  it("renders", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
