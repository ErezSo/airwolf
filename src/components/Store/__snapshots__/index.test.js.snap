// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store renders correctly 1`] = `
"<div style={{...}}>
  <div style={{...}}>
    <Item item={{...}} idx={0} reserveItem={[Function: mockConstructor]} />
    <Item item={{...}} idx={1} reserveItem={[Function: mockConstructor]} />
    <Item item={{...}} idx={2} reserveItem={[Function: mockConstructor]} />
    <Item item={{...}} idx={3} reserveItem={[Function: mockConstructor]} />
    <Item item={{...}} idx={4} reserveItem={[Function: mockConstructor]} />
    <Item item={{...}} idx={5} reserveItem={[Function: mockConstructor]} />
    <Item item={{...}} idx={6} reserveItem={[Function: mockConstructor]} />
    <Item item={{...}} idx={7} reserveItem={[Function: mockConstructor]} />
  </div>
</div>"
`;
